### PR TITLE
Support `listIndexes` command

### LINKED ISSUE #278: Support `listIndexes` command
Implement `listIndexes` command stub that returns valid empty results in the same format as MongoDB.

Ignore `cursor` and `comment` fields.

https://www.mongodb.com/docs/manual/reference/command/listIndexes/

### Definition of Done

- all handlers updated;
- integration/compatibility tests added/updated (results for compat/target might be different):
   - collection doesn't exist;
   - existing collection; 
- spot refactorings done;
- user documentation updated.

### PR DESCRIPTION
**Pull request by rumyantseva**
# Description

Closes #278.

## Readiness checklist

* [ ] I added/updated unit tests.
* [x] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [x] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: integration/commands_administration_test.go [hunk 1]
```diff
@@ -986,3 +986,81 @@ func TestCommandsAdministrationCurrentOp(t *testing.T) {
...
+
+	for name, tc := range map[string]struct { //nolint:vet // for readability
+		collectionName any
+		expectedResult bson.D
+		expectedError  *mongo.CommandError
+		skip           string
+	}{
+		"existing-collection-default-indexes": {
+			collectionName: collection.Name(),
+			expectedResult: bson.D{
```
Comment (w84thesun) [inline integration/commands_administration_test.go:1003]: Can we just skip comparing `firstBatch` field for now?

**Review by w84thesun — COMMENTED** [review]
Nice one! Added some comments.

File: integration/commands_administration_test.go (hunk 1 above)
Comment (w84thesun) [reply integration/commands_administration_test.go:1003]: We can use `ConvertDocument` helper to get `types.Document` and just check expected fields.

Comment (rumyantseva) [reply integration/commands_administration_test.go:1003]: Ah, great! Thanks!

File: integration/commands_administration_test.go [hunk 2]
```diff
@@ -986,3 +986,81 @@ func TestCommandsAdministrationCurrentOp(t *testing.T) {
...
+	} {
+		name, tc := name, tc
+
+		t.Run(name, func(t *testing.T) {
+			if tc.skip != "" {
+				t.Skip(tc.skip)
+			}
+
+			var res bson.D
+			err := collection.Database().RunCommand(
```
Comment (AlekSi) [inline integration/commands_administration_test.go:1051]: Let's test `collection.Indexes()`

Comment (rumyantseva) [reply integration/commands_administration_test.go:1055]: That produces a data race on the driver side 😅 
I pushed the code that demonstrates it.

Comment (rumyantseva) [reply integration/commands_administration_test.go:1053]: Added `Indexes().List()` for a compat test.

File: integration/commands_administration_test.go [hunk 3]
```diff
@@ -986,3 +986,81 @@ func TestCommandsAdministrationCurrentOp(t *testing.T) {
 	_, ok := must.NotFail(doc.Get("inprog")).(*types.Array)
 	assert.True(t, ok)
 }
+
+func TestCommandsAdministrationListIndexes(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (rumyantseva) [reply integration/commands_administration_test.go:990]: Fixed!

File: internal/handlers/common/listindexes.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+)
+
+// ListIndexesParams contains `listIndexes` command parameters.
+type ListIndexesParams struct {
+	DB         string
+	Collection string
+}
+
+// GetListIndexesParams returns `listIndexes` command parameters.
+func GetListIndexesParams(document *types.Document, l *zap.Logger) (*ListIndexesParams, error) {
```
Comment (AlekSi) [inline internal/handlers/common/listindexes.go:33]: Let's remove that: all index-related methods do not receives tens of parameters like `find`

Comment (rumyantseva) [reply internal/handlers/common/listindexes.go:33]: Well, 24 lines of code that are absolutely the same for all the current and future handlers. But if it matters, I can copy-paste, yes.

Comment (AlekSi) [reply internal/handlers/common/listindexes.go:33]: For better or worse, we currently do that only for commands with a lot of parameters. Doing that there while keeping, for example, `insert` separate would make our code base less consistent

Comment (rumyantseva) [reply internal/handlers/common/listindexes.go]: That would be nice to know how much "a lot" is. So, four (the current case) is not a lot. Is five a lot?

**Review by w84thesun — APPROVED** [review]
👍

**Review by noisersup — APPROVED** [review]
:100:

**Review by Fashander — APPROVED** [review]

File: internal/handlers/common/listindexes.go (hunk 4 above)
Comment (AlekSi) [reply internal/handlers/common/listindexes.go]: That's the best answer I can provide: https://www.youtube.com/watch?v=iG2YxzjC9WI 😆 

Seriously, though, that's not about the number itself, but about consistency

File: integration/commands_administration_test.go [hunk 5]
```diff
@@ -986,3 +986,75 @@ func TestCommandsAdministrationCurrentOp(t *testing.T) {
...
+			// TODO Use simple assert.Equal after https://github.com/FerretDB/FerretDB/issues/1384
+			// assert.Equal(t, compatRes, targetRes)
+			assert.Empty(t, targetRes)
+			assert.NotEmpty(t, compatRes)
+		})
+	}
+}
+
+// TestCommandsAdministrationRunCommandListIndexes tests the behavior when listIndexes is called through RunCommand.
+// It's handy to use it to test the correctness of errors.
```
Comment (AlekSi) [inline integration/commands_administration_test.go:1020]: I don't think we should test through `RunCommand` in general: we want compatibility with apps through drivers, not with specs/docs via `RunCommand`. There are some exceptions like `drop` that hide some errors, but they are exceptions. There is no need to over-test simple commands like `listIndexes`

**Pull request merged**